src/main/webapp/scripts/script.js

```diff
@@ -1,18 +1,23 @@
-const API_KEY = 'AIzaSyAr7MKPZKUCQ9G-N00I44EJqvsYjE3jAqU';
+const API_KEY = '';
 const YOUTUBE_DISCOVERY = 'https://www.googleapis.com/discovery/v1/apis/youtube/v3/rest';
 
-function loadApi(callback) {
-  gapi.load('client', function() {
-    gapi.client.init({
-      'apiKey': API_KEY,
-      'discoveryDocs': [YOUTUBE_DISCOVERY]
-    })
-    .then(function() {
-      callback();
-    })
-    .catch((error) => {
-      updateDom('Oops! An error occured with Youtube, please try again later.', 'popular-list-container');
-      console.log('Was not able to load Google API client:', error);
+/**
+ * Loads any Google APIs listed in the discoveryDocs
+ * @return {Promise} for completing the load
+ */
+function loadApi() {
+  return new Promise((resolve, reject) => {
+    gapi.load('client', () => {
+      gapi.client.init({
+        'apiKey': API_KEY,
+        'discoveryDocs': [YOUTUBE_DISCOVERY]
+      })
+      .then(resolve)
+      .catch((error) => {
+        alert('Oops! An error occured with Youtube, please try again later.');
+        throw error;
+        reject();
+      });
```

Comment:
I believe exceptions are converted to rejects automatically. So no need to catch if all you are going to do is rethrow and reject.